infra/docker/server.Dockerfile

```
@@ -4,32 +4,32 @@ RUN apk add --no-cache openssl && \
 
-# Copy source code
+# 3. NOW copy everything else (including your schema and source code)
 COPY . .
 
-# Build shared types, generate prisma client, and build server (single layer)
+# 4. Manually run the generation and builds now that all files are present
 ARG DATABASE_URL="postgresql://placeholder:placeholder@localhost:5432/placeholder"
-RUN pnpm --filter @shared/types build && \
-    cd server && npx prisma generate && \
-    cd .. && pnpm --filter server build
+RUN cd server && npx prisma generate && \
```

Hadolint (2.14.0)

[warning] Use WORKDIR to switch to a directory

(DL3003)